rippled v1.3.1 Docs

mDuo13 opened the pull request:
- Updated install/upgrade instructions, including migration information for 1.2.x to 1.3.x
- fixMasterKeyAsRegularKey amendment description
- log rotation stuff

This stuff has all been reviewed individually. Link checker will fail until the 1.3.1 tag exists in the `rippled` repository.

_Edit: Changed to 1.3.1 instead of 1.3.0._

mDuo13 committed:
Move Ubuntu install location

mDuo13 committed:
rippled 1.3.0 install instructions (DRAFT)

mDuo13 committed:
Add rippled v1.3.0 release link

alloynetworks committed:
Update logrotate.md

Starting with `rippled` 1.3, the `/etc/logrotate.d/rippled` script will be installed by the DEB and RPM packages. Modifications made to reflect that.

mDuo13 committed:
more 1.3.0 update/install/migration work

mDuo13 committed:
v1.3.0 install/upgrade (draft done?)

mDuo13 committed:
Sidebar: hide subpages with redirect template

mDuo13 committed:
rippled v1.3.0 install/upgrade/build updates

mDuo13 committed:
Add fixMasterKeyAsRegularKey amendment details

mDuo13 committed:
Update install instructions per reviews

mDuo13 committed:
'configuration file' → 'config file'

mDuo13 committed:
Corrections for v1.3.0 automatic updates

mDuo13 committed:
'Black hole' is two words

mDuo13 commented:
Rebased on latest `master`

intelliot commented on content/tutorials/manage-the-rippled-server/installation/build-run-rippled-macos.md:
```diff
@@ -24,7 +24,7 @@ For development purposes Ripple recommends running `rippled` as your own user, n
         $ brew install git cmake pkg-config protobuf openssl ninja
 
-0. Install Boost 1.67.0. `rippled` 1.2.0 is compatible with Boost 1.67. <!--#{ no boost@1.67 formula, so must manually compile and install. will eventually upgrade to boost 1.68, but needs testing first }# -->
+0. Install Boost 1.67.0 or higher. `rippled` 1.3.0 is compatible with Boost 1.67.0 and higher. The latest version of Boost available in the Homebrew repositories isn't new enough, so you must install Boost manually:
```
Is it compatible with Boost 1.67.0? I got this error:
```
...
-- rippled version: 1.3.0
-- NIH-EP cache path: /Users/user/ripple/rippled/.nih_c/unix_makefiles/AppleClang_10.0.0.10001145/Debug
-- using [caa5c9e2232ff1eee1d0c61907283649e48e6fbc] as build container tag...
-- BOOST_ROOT: /Users/user/Downloads/boost_1_67_0
-- BOOST_LIBRARYDIR: 
CMake Error at Builds/CMake/FindBoost.cmake:2153 (message):
  Unable to find the requested Boost libraries.

  Boost version: 1.67.0

  Boost include path: /Users/user/Downloads/boost_1_67_0

  Detected version of Boost is too old.  Requested version was 1.70 (or
  newer).
Call Stack (most recent call first):
  CMakeLists.txt:717 (find_package)


CMake Error at CMakeLists.txt:1114 (get_target_property):
  get_target_property() called with non-existent target "Boost::date_time".


-- Pausing to download NuDB...
-- Could NOT find Doxygen (missing: DOXYGEN_EXECUTABLE) 
-- doxygen executable not found -- skipping docs target
-- Configuring incomplete, errors occurred!
...
```

I'm building Boost 1.70.0 now, to give it a try. Using:
```
./b2 cxxflags="-std=c++14" visibility=global
```
(The `visibility=global` is new.)

mDuo13 committed:
rippled 1.3.0 requires Boost 1.70 (as of rc2)

mDuo13 committed:
RocksDB not supported for history sharding on 1.3

- Updated troubleshooting with example error
- Removed stale RocksDB-on-shards error from troubleshooting
- Updating configuration tutorial to mention that NuDB & SSD are
  required for shards

mDuo13 committed:
130 shard updates: edits per reviews

mDuo13 committed:
Merge pull request #628 from mDuo13/no_130_rocks_shards

RocksDB not supported for history sharding on 1.3

alloynetworks committed:
Missing install keyword

`apt -y apt-transport-https ca-certificates wget gnupg` changed to `apt -y install apt-transport-https ca-certificates wget gnupg`

mDuo13 committed:
1.3.1 supersedes 1.3.0

mDuo13 committed:
Fix broken link to '1.3.0' migration instructions

mDuo13 committed:
Merge pull request #634 from alloyxrp/patch-1

Missing install keyword
